src/dataset.jl

```diff
@@ -172,16 +172,133 @@ end
+    else
+        error("Could not determine dataset type (raster or vector).")
+    end
+    return nothing
+end
+
+"""
+    writerasters(dataset::AbstractDataset, filename::AbstractString; kwargs...)::Nothing
+
+Write a raster dataset to disk.
+
+### Parameters
+* `dataset`       the dataset to write
+
+### Keyword Arguments
+* `filename`      the filename, UTF-8 encoded.
+* `driver`        the driver to use
+* `strict`        ``true`` if the copy must be strictly equivalent, or more
+normally ``false`` if the copy may adapt as needed for the output format.
+* `options`       additional format dependent options controlling creation
+of the output file. `The APPEND_SUBDATASET=YES` option can be specified to
+avoid prior destruction of existing dataset.
+
+### Returns
+nothing
+"""
+function writerasters(dataset::AbstractDataset,
+                      filename::AbstractString;
+                      kwargs...)::Nothing
     destroy(unsafe_copy(dataset, filename = filename; kwargs...))
     return nothing
 end
 
+"""
+    writelayers(ds, filename; driver, driver_options=[""], layer_options=[""], chunksize=20000)
+
+Writes the vector dataset to the designated filename. The options are passed to the newly created dataset and
+have to be given as a list of strings in KEY=VALUE format. The chunksize controls the number of features written 
+in each database transaction, e.g. for SQLite. This function can also be used to copy datasets on disk.
+
+Currently working drivers: FlatGeobuf, GeoJSON, GeoJSONSeq, GML, JML, KML, MapML, ESRI Shapefile, SQLite
+GPKG should mostly work, too
+
+### Parameters
+*`dataset`  The source dataset
+*`filename` The file name to write to
+
+### Keyword arguments
+`driver`           The driver to use, you have to manually select the right driver for the file extension you wish
+`driver_options`   A vector of strings containing KEY=VALUE pairs for driver-specific creation options
+`layer_options`    Driver specific options for layer creation
+`chunksize`        Number of features to write in one database transaction. Neglected when `use_gdal_copy` is true.
+`use_gdal_copy`    Set this to true if you encounter errors when writing. Set to false for higher speed.
+
+### Returns
+nothing
+"""
+function writelayers(dataset::AbstractDataset,
+                     filename;
+                     driver=getdriver(dataset),
+                     driver_options::Vector{String}=[""], 
+                     layer_options::Vector{String}=[""],
+                     chunksize=20000,
+                     use_gdal_copy=false)
+    create(filename; driver=driver, options=driver_options, width=0, height=0, nbands=0) do target
+        for layeridx in 0:nlayer(dataset)-1
+            sourcelayer = getlayer(dataset, layeridx)
+            sourcelayerdef = layerdefn(sourcelayer)
+
+            if use_gdal_copy
+                unsafe_copy(sourcelayer; dataset=target, name=getname(sourcelayer))
```

Comment:
Why is `unsafe_copy` being used here? Do we need to `destroy` the object that's returned by it?